File: fe2-android/app/src/test/framework/robolectric/java/com/github/dedis/popstellar/di/AppDatabaseModuleHelper.java
package com.github.dedis.popstellar.di;

import android.content.Context;

import androidx.room.Room;

import com.github.dedis.popstellar.repository.database.AppDatabase;
import com.github.dedis.popstellar.repository.database.CustomTypeConverters;

public class AppDatabaseModuleHelper {

  public static AppDatabase getAppDatabase(Context context) {
    return Room.inMemoryDatabaseBuilder(context, AppDatabase.class)
        .allowMainThreadQueries()
        .fallbackToDestructiveMigration()
        .addTypeConverter(
            new CustomTypeConverters(
                JsonModule.provideGson(DataRegistryModule.provideDataRegistryForGson())))
        .build();
  }
}

File: fe2-android/app/src/test/framework/robolectric/java/com/github/dedis/popstellar/di/AppDatabaseModuleHelper.kt
package com.github.dedis.popstellar.di

import android.content.Context
import androidx.room.Room.inMemoryDatabaseBuilder
import com.github.dedis.popstellar.repository.database.AppDatabase
import com.github.dedis.popstellar.repository.database.CustomTypeConverters

object AppDatabaseModuleHelper {
  @JvmStatic
  fun getAppDatabase(context: Context): AppDatabase {
    return inMemoryDatabaseBuilder(context, AppDatabase::class.java)
      .allowMainThreadQueries()
      .fallbackToDestructiveMigration()
      .addTypeConverter(
        CustomTypeConverters(
          JsonModule.provideGson(DataRegistryModule.provideDataRegistryForGson())
        )
      )
      .build()
  }
}
